### DISCUSSION TITLE
is is possible to scale a lwtk application?

### DISCUSSION THREAD

**Discussion by mark-summerfield** [osch/lua-lwtk#9] [Q&A] [answered]
I'm trying out some of the examples using lua 5.4.
1. How can I set the font size for each type of widget?
2. How can I  *scale* the whole application? For example, on my laptop the screenshot actually looks much smaller than github shows.
![screenshot1](https://github.com/user-attachments/assets/895644a1-acf2-46e9-933a-85fafe30049f)

**Comment by osch** [comment, marked as answer]
You can scale the whole lwtk application by setting  the `scaleFactor` style parameter. This can be done by setting this style parameter somewhere in the style hierarchy. This can be done for the whole application for example in [example01.lua](https://github.com/osch/lua-lwtk/blob/master/example/example01.lua) replace

```lua
local app = Application("example01.lua")
```

with 

```lua
local app = Application("example01.lua", { scaleFactor = 3 })
```

or you can set this style parameter for an individual widget subtree, see the documentation for [lwtk.Style](https://github.com/osch/lua-lwtk/blob/master/doc/Style.md).

Anyway, there should be some support for hardware dependent screen scale factor, so my question is:

What does 
```lua
print(app:getScreenScale())
```

give on your platform? 

However, if you only want to increase the font size and not the whole scale factor and not providing a totally new style there is up to now no easy solution. I will try to provide a patch for this for a better solution.

**Comment by osch** [comment]
Update: with commit 429cfe9 you could replace

```lua
local app = Application("example01.lua")
```

with 

```lua
local app = Application("example01.lua", { TextSize = 20})
```
for overriding the `TextSize` style parameter for all widgets.

**Comment by mark-summerfield** [comment]
Thank you.
